### DISCUSSION TITLE
Ast output

### DISCUSSION THREAD

**Discussion by olup** [vlang/v#7628] [General]
Hi again.

I was wondering if I could help with the vscode plugin and I wanted to scaffold a simple language server to present user with live errors and warnings from compilation in the files themselves.
For that I need access to structured data about said errors and warning, things that should - If I understand correctly - be part of the AST.

There is a resolved issue where it's announced that an AST has been implemented, is there a command to output it ?

There is also a library called vast that is supposed to do just that but I can't say if it has been updated to 0.2 - I could compile it earlier today.

Error highlight and autocomplete / Intellisense is must-have, especially for v beginers as myself.

**Comment by medvednikov** [comment]
Hi

We already have an actively developed LS: https://github.com/vlang/vls

Contributions are welcome :)

Please join Discord channel #vls

**Reply by olup** [reply]
Many thanks, I use the vscode extension already so I know about the active language server. I actually wanted to see how hard it would be to contribute to it, and also know about the AST thing.

**Reply by medvednikov** [reply]
@olup I've created a feature request for that

https://github.com/vlang/v/issues/8158
